### ISSUE TITLE
WriteFile bug

### ISSUE THREAD

**Issue by badrelmers** [bitfield/script#81]
the doc says that WriteFile() is equivalent to > . 
but when I use WriteFile() with a file which contains content already, it will not truncate the file but will append it to the begining! overwriting only the bytes I write.
`cat file.txt`
result: ttttttttttt

`script.Echo("zzz").WriteFile("file.txt")`
result: zzztttttttt

I think we have to use O_TRUNC here 
https://github.com/bitfield/script/blob/32cf7a77c523547622185ce6fb5ee6d787ec4d42/sinks.go#L112

tested in windows 7 with golang  v1.16.2

**Comment by bitfield** [comment]
Thanks! I'm sure you're right. The weird thing is, this works as expected on MacOS. I don't know enough about the respective kernels to know why this would be, but adding `O_TRUNC` to the flags doesn't change the behaviour at all. If it fixes your problem on Windows, I see no reason not to make this change. I'll run up a PR for it.

**Comment by lyderic** [comment]
I don't think that it's a Windows only problem. I am on Linux (ubuntu LTS 20.04, kernel 5.11, go 1.17.1) and I have the same behaviour as the OP.

Example program to reproduce:

```
package main

import (
        "github.com/bitfield/script"
)

func main() {
        script.Echo("abcdef").WriteFile("foo")
        script.Echo("bar").WriteFile("foo")
        script.File("foo").Stdout()
}

```

The output of the program is '**bardef**', whereas I would expect the output to be '**bar**'.

Many thanks for looking into this and also many thanks for the _script_ library!

**Comment by bitfield** [comment]
It must be working accidentally because I've only been able to test it on macOS, then! Thanks for the report @lyderic.

**Comment by lyderic** [comment]
Hi John,

I confirm that the fix works for me (Linux Arch, go 1.17.5, 'fix_truncate_bug_81' branch).

Many thanks for the fix.

Cheers,
Lyderic
